### DISCUSSION TITLE
data.jointTorqueRegressor is transposed

### DISCUSSION THREAD

**Discussion by muttistefano** [stack-of-tasks/pinocchio#2522] [Q&A]
Hi all,
It seems that when i use the computeJointTorqueRegressor, the result stored in data.jointTorqueRegressor is transposed.
In the code, I receive a [joint number x number of parameters], while in literature, it is the other way around.
Is there a reason it is implemented like this?
Thanks

**Comment by jcarpent** [comment]
Could you provide some references? Is the documentation clear enough?

At first glance, this is an implementation choice.

**Reply by muttistefano** [reply]
Hi, in siciliano's Robotics Modelling, Planning, and Control (https://nibmehub.com/opac-service/pdf/read/Robotics%20Modelling-%20Planning%20and%20Control%20by%20Bruno%20Siciliano-.pdf)

At the end of section 7.2.2  
![img](https://github.com/user-attachments/assets/a8f9d7d3-51cf-4ab4-8e5c-c47cdc0bde16)

Indeed, it might be an implementation choice, not a problem.

**Reply by muttistefano** [reply]
i guess it was my mistake, the matrix is correctly shaped
